src/core/chuck_vm.cpp

```diff
@@ -1155,8 +1192,10 @@ Chuck_VM_Shred * Chuck_VM::spork( Chuck_VM_Shred * shred )
     shred->now = shred->wake_time = m_shreduler->now_system;
     // set the id, if one hasn't been assigned yet | 1.5.0.8 (ge) add check
     if( !shred->xid ) shred->xid = next_id();
+    // check if need to raise | 1.5.1.4
+    if( !m_shred_check4dupes && shred->xid > m_shred_id ) m_shred_id = shred->xid;
```

Comment:
added this comment (and you as a blame-worthy author hah)
```
        // check if we need to raise m_shred_id as highest shred ID...
        // NOTE if check4dupes is enabled, m_shred_id may no longer be the
        // highest ID but instead is a counter for possible next shred IDs,
        // which will repeatedly increment+check for duplicates until an
        // available ID is found | 1.5.1.4 (ge and nshaheed) added
        if( !m_shred_check4dupes && shred->xid > m_shred_id )
```
also good question about where this logic should live; I've moved it into `next_id()` so that the intricacies of `m_check4dupes` is contained within `next_id()` (plus initialization and a reset case)